src/secret_service_with_keyutils.rs

```diff
@@ -0,0 +1,169 @@
+/*!
+
+# secret-service-with-keyutils credential store
+
+TODO
+
+ */
+use std::collections::HashMap;
+
+#[cfg(feature = "sync-secret-service")]
+use dbus_secret_service::{Error, Item};
+#[cfg(feature = "async-secret-service")]
+use secret_service::{blocking::Item, Error};
+
+#[cfg(all(target_os = "linux", feature = "linux-native"))]
+use crate::keyutils::KeyutilsCredential;
+use crate::secret_service::{decode_error, SsCredential};
+
+use super::credential::{Credential, CredentialApi, CredentialBuilder, CredentialBuilderApi};
+use super::error::Result;
+
+#[derive(Debug, Clone)]
+pub struct SsKeyutilsCredential {
+    keyutils: KeyutilsCredential,
+    ss: SsCredential,
+}
+
+impl CredentialApi for SsKeyutilsCredential {
+    fn set_password(&self, password: &str) -> Result<()> {
+        self.set_secret(password.as_bytes())
+    }
+
+    fn set_secret(&self, secret: &[u8]) -> Result<()> {
+        self.ss.set_secret(secret)?;
+        let _ = self.keyutils.set_secret(secret);
+        Ok(())
+    }
+
+    fn get_password(&self) -> Result<String> {
+        if let Ok(password) = self.keyutils.get_password() {
+            return Ok(password);
+        }
+
+        let password = self.ss.get_password()?;
```

Comment:
This passes errors from the secret service directly to the client, which you can't do, because the secret service the Ambiguous error, which this keystore must not (see comments below), and also because that Ambiguous error would contain secret-service credentials, not credentials from this service (see comments below). You have to analyze the error returned and return an appropriate error instead, which should only be errors that don't return credentials.